internal/generic/generic.go

```diff
@@ -0,0 +1,154 @@
+/*
+Package generic provides a set of generic helper functions,
+primarily aimed at common channel operations.
+*/
+package generic
```

Comment:
Most of the code in this package isn't called yet. It shouldn't be included until it's actually required by calling code.